src/main/java/edu/cornell/kfs/sys/batch/MoveFileStep.java

```diff
@@ -0,0 +1,128 @@
+package edu.cornell.kfs.sys.batch;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.sys.batch.AbstractStep;
+
+import edu.cornell.kfs.sys.CUKFSConstants;
+
+/**
+ * Generic batch step for moving files from one directory to another.
+ * 
+ * At a minimum, the sourcePath and targetPath must be set to valid pre-existing directories,
+ * and the fileNamePattern must be set to a valid Java regex String. Only files whose names
+ * match the regex will be moved from the source directory to the target directory.
+ * This step will also automatically create ".done" files in the target directory for each moved file.
+ * 
+ * The source/target directory paths are not required to have trailing slashes. This step
+ * will automatically add them as needed.
+ */
+public class MoveFileStep extends AbstractStep {
+
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(MoveFileStep.class);
+
+    protected String sourcePath;
+    protected String targetPath;
+    protected Pattern fileNamePattern;

```

Comment:
My goal was to make this step generic enough so that multiple instances of this class could be used to move the files accordingly, given that the source and target directories would be different for each. (Also, this step may be needed for the request-extract-to-PDP file move.) That way, you could have multiple beans representing a specific configured version of this step.

If you need me to, I can try to implement multiple source-to-target behavior for a single step instance, but I think that would introduce too much complexity.